packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -620,24 +639,44 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
                     throw new Error(`Invalid snapshot format version ${snapshotFormatVersion}`);
                 }
                 this.pkg = pkgFromSnapshot;
-                isRootStore = isRootDataStore;
             }
 
-            /**
-             * If there is no isRootDataStore in the attributes blob, set it to true. This will ensure that
-             * data stores in older documents are not garbage collected incorrectly. This may lead to additional
-             * roots in the document but they won't break.
-             */
             this.details = {
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 pkg: this.pkg!,
-                isRootDataStore: isRootStore ?? true,
                 snapshot: tree ?? undefined,
+                // back-compat - 0.30. This has been moved to IGCDetails.
+                isRootDataStore: true,
             };
         }
 
         return this.details;
     }
+
+    protected async getInitialGCDetails(): Promise<IGCDetails> {
+        if (this.initialGcDetails === undefined) {
+            /**
+             * Default value of initial GC details in case the initial snapshot does not have GC details blob.
+             * isRootNode is set to true so that data stores in older documents are not collected by GC.
+             */
+            let gcDetails: IGCDetails = {
```

Comment:
it feels like you can get rid of this variable and directly modify this.initialGcDetails